### PR TITLE
xfstests: Partition device size will use more free space

### PR DESCRIPTION
**Pull request by SeroSun**
Modify partition part. Use more free space than beforej.

- Related ticket: https://progress.opensuse.org/issues/37895
- Verification run: http://10.67.133.102/tests/988

### CODE REVIEW

**Comment by lansuse** [comment]
LGTM

**Comment by SeroSun** [comment]
New verify run is: http://10.67.133.102/tests/988
Could see from http://10.67.133.102/tests/988#step/partition/13, it use as much free space as before.

**Comment by lansuse** [comment]
LGTM

File: tests/xfstests/partition.pm [hunk 1]
```diff
@@ -18,18 +18,69 @@ use base 'opensusebasetest';
 use utils;
 use testapi;
 
+sub str_to_mb {
+    my $str = shift;
```
Comment (jlausuch) [inline tests/xfstests/partition.pm]: Another option is `my ($str) = @_;`

Comment (SeroSun) [reply tests/xfstests/partition.pm]: I see, I wonder is there any benefit to use @_ than shift :)

Comment (cfconrad) [reply tests/xfstests/partition.pm]: From my perspective: It avoid using implicit variables (but it's quite common in perl) and  you could use multiple parameters in one command.

But in this specific case, I do not see a benefit.

Comment (SeroSun) [reply tests/xfstests/partition.pm]: Thanks! I totally understand now.

**Comment by ostruzina** [comment]
Hello @SeroSun, after this change has been merged, we have noticed xfstests* failing heavily in QAM tests:
https://openqa.suse.de/tests/overview?distri=sle&flavor=Server-DVD-Updates&version=15&groupid=165&build=20190612-1

https://openqa.suse.de/tests/2968846#step/partition/13

Could you take a look please?

**Comment by jlausuch** [comment]
@ostruzina if the issue is critical and blocking, we can revert it immediately.

**Comment by SeroSun** [comment]
@ostruzina @jlausuch  that problem caused by https://github.com/SUSE/qa-testsuites/pull/746, I have revert that one.

**Comment by SeroSun** [comment]
@ostruzina @jlausuch I have retriggered xfstests in build=20190612-1 and build=20190613-1. Thank you for info me that problem.

**Comment by SeroSun** [comment]
@ostruzina retrigger is not enough, could you please do some configure modify? Yesturday I sent an email to pcervinka, Jan Baier and Ondrej, told about setting changes in xfstests. I'll copy content here.

As changes in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/7647
some configure need to modify:

1. Added a new test "create_hdd_xfstests"
BOOT_HDD_IMAGE=1
DESKTOP=textmode
HDD_1=SLES-%VERSION%-%ARCH%-%BUILD%@%MACHINE%-
minimal_with_sdk%BUILD_SDK%_installed_withhome.qcow2
MAX_JOB_TIME=32400
NO_KDUMP=0
PUBLISH_HDD_1=SLES-%VERSION%-%ARCH%-%BUILD%@%MACHINE%-
minimal_with_sdk%BUILD_SDK%_xfstests.qcow2
QA_HEAD_REPO=http://149.44.176.2/ibs/QA:/Head/SLE-%VERSION%
VIRTIO_CONSOLE=1
XFSTESTS=installation
START_AFTER_TEST=create_hdd_minimal_base+sdk_withhome

2. Change sub tests xfstests_*:  START_AFTER_TEST fromm create_hdd_minimal_base+sdk_withhome to create_hdd_xfstests
  START_AFTER_TEST=create_hdd_xfstests

3. Change HDD_1 as following
  HDD_1=SLES-%VERSION%-%ARCH%-%BUILD%@%MACHINE%-
minimal_with_sdk%BUILD_SDK%_xfstests.qcow2

**Pull request merged**